Audit Log Documentation for #262

msciotti opened the pull request:
Also includes some related changes, like documentation for `default_message_notification_level` and `explicit_content_filter_level`

Rapptz commented on docs/resources/Guild.md:
```diff
@@ -460,3 +475,127 @@ Returns the [guild embed](#DOCS_GUILD/guild-embed-object) object. Requires the '
+| id | snowflake | id of the entry
+| action_type | an [audit log event](#DOCS_GUILD/audit-log-event) | type of action that occured
+| options | object | additional information specific to the action type
+
```
This is missing the `reason` field.

Rapptz commented on docs/resources/Guild.md:
```diff
@@ -460,3 +475,127 @@ Returns the [guild embed](#DOCS_GUILD/guild-embed-object) object. Requires the '
+| user_id | snowflake | the user who made the changes
+| id | snowflake | id of the entry
+| action_type | an [audit log event](#DOCS_GUILD/audit-log-event) | type of action that occured
+| options | object | additional information specific to the action type
```
Not sure how you would expand on this but the `options` key is used in the following places:

1. `MEMBER_PRUNE` has a `delete_members_days` key as a `string` value with how far the prune went and a `members_removed` key (again with a `string` value) with how many members were pruned.
2. `MESSAGE_DELETE` has two keys, `channel_id` snowflake for where the message got deleted and `count` string for how many messages got deleted.
3. `CHANNEL_OVERWRITE_[CREATE|UPDATE|DELETE]` have an `id` snowflake with the overwrite that got changed and a `type` that says the overwrite type (which can be `member` or `role`). If the `type` is equal to `role` then there is another key, `role_name` which obviously has the role name.

Rapptz commented on docs/resources/Guild.md:
```diff
@@ -460,3 +475,127 @@ Returns the [guild embed](#DOCS_GUILD/guild-embed-object) object. Requires the '
+| bitrate | int | voice channel bitrate changed
+| permission_overrides | array of [channel overwrite](#DOCS_CHANNEL/overwrite-object) objects| permissions on a channel changed |
+| $add | [role]() object| new role added
+| $remove | [role]() object | role removed
```
These are missing the link.

Rapptz commented on docs/resources/Guild.md:
```diff
@@ -460,3 +475,127 @@ Returns the [guild embed](#DOCS_GUILD/guild-embed-object) object. Requires the '
+| allow | int | a permission on a text or voice channel was allowed for a role
+| deny | int | a permission on a text or voice channel was denied for a role
+| nsfw | bool | channel nsfw restriction changed
+| reason | string | the reason for the change in the audit log
```
This isn't in the change object but in the entry object.

Rapptz commented on docs/resources/Guild.md:
```diff
@@ -460,3 +475,127 @@ Returns the [guild embed](#DOCS_GUILD/guild-embed-object) object. Requires the '
+| permissions | int | [permissions](#DOCS_PERMISSIONS/permissions-bitwise-permission-flags) for a role changed
+| color | int | role color changed
+| hoist | bool | role is now displayed/no longer displayed separate from online users
+| mentioning | bool | role is now mentionable/unmentionable
```
This is `mentionable` not `mentioning`.

Rapptz commented on docs/resources/Guild.md:
```diff
@@ -460,3 +475,127 @@ Returns the [guild embed](#DOCS_GUILD/guild-embed-object) object. Requires the '
+| topic | string | text channel topic changed
+| type | string/int | type of entity created - can be of [channel type](#DOCS_CHANNEL/channel-types) or a string like "role"|
+| bitrate | int | voice channel bitrate changed
+| permission_overrides | array of [channel overwrite](#DOCS_CHANNEL/overwrite-object) objects| permissions on a channel changed |
```
This is `permission_overwrites` not `permission_overrides`

msciotti replied:
Interesting. Bug/overlap in the code somewhere, as it is denoted `permission_overrides`

msciotti replied:
I'll swap it to `overwrites` as that's what comes in the response object, and will investigate the discrepency

Rapptz commented on docs/resources/Guild.md:
```diff
@@ -460,3 +475,138 @@ Returns the [guild embed](#DOCS_GUILD/guild-embed-object) object. Requires the '
+| ---- | ----- | ---------- |
+| user_id | snowflake | filter the log for a user id |
+| action_type | int  | the type of [audit log event](#DOCS_GUILD/audit-log-events)|
+| before | snowflake | filter the log before a certain action id |
```
I think `before a certain audit log entry id` might be better.

jhgg commented on docs/resources/Guild.md:
```diff
@@ -460,3 +475,138 @@ Returns the [guild embed](#DOCS_GUILD/guild-embed-object) object. Requires the '
+| key | string | type of audit log [change key](#DOCS_GUILD/audit-log-change-key)
+
+###### Audit Log Change Key
+| Name | Value Type | Description |
```
we should clarify maybe which object, each key corresponds to.
```
| name | guild | string | name changed 
| topic | text channel | string | etc |
```

msciotti replied:
Good thought. I can link the objects that way as well

jhgg commented on docs/resources/Guild.md:
```diff
@@ -160,6 +160,21 @@ Represents an Offline Guild, or a Guild whose information has not been provided
 | managed | bool | whether this emoji is managed |
 
+###### Default Message Notification Level
+|Key|Value|
```
weird spacin here.

msciotti replied:
¯\\\_(ツ)\_/¯ Markdown knows what I mean. But I can go through and make sure all the raw spacing is perty before merging

abuneri commented on docs/resources/Audit_Log.md:
```diff
@@ -0,0 +1,143 @@
+| explicit_content_filter | [guild](#DOCS_GUILD/guild-object) | int | change in [whose messages](#DOCS_GUILD/explicit-content-filter-level) are scanned and deleted for explicit content in the server
+| default_message_notifications | [guild](#DOCS_GUILD/guild-object) | int | default [message notification level](#DOCS_GUILD/default-message-notification-level) changed|
+| vanity_url_code | [guild](#DOCS_GUILD/guild-object) | string | guild invite vanity url changed
+| $add | [guild](#DOCS_GUILD/guild-object) | [role](#DOCS_PERMISSIONS/role-object) object| new role added
```
The value sent for the change keys `$add` and `$remove` is `[{role_obj}]` not just `{role_obj}`. Unless that has changed, should probably be clarified.

msciotti replied:
Correct, thank you

SEAPUNK commented on docs/resources/Audit_Log.md:
```diff
@@ -0,0 +1,221 @@
+| max_uses | [invite](#DOCS_INVITE/invite-metadata-object) | int | change to max number of times invite code can be used |
+| uses | [invite](#DOCS_INVITE/invite-metadata-object) | int | number of times invite code used changed |
+| max_age | [invite](#DOCS_INVITE/invite-metadata-object) | int | how long invite code lasts changed |
+>>>>>>> fb5549435c9c8b10a0ade96d68bea3921ea496d5
```
you got some broken merge stuff here and in other places

SEAPUNK reviewed (dismissed):
fix merge artifacts pls

msciotti committed:
Fixed weird master rebase back to normal

msciotti committed:
Merge branch 'master' into feature/AuditLogDocumentation

jhgg commented on docs/resources/Audit_Log.md:
```diff
@@ -0,0 +1,143 @@
+
+### Audit Log Object
+
+###### Audit Log Object Structure
```
Uhhm. This would really be the /response/ structure. Not the object structure. Right?

msciotti replied:
Changing to "Audit Log Structure" to match the rest of our docs. Channel, for example, is "channel structure". Guild and message are similar

jhgg commented on docs/resources/Audit_Log.md:
```diff
@@ -0,0 +1,143 @@
+| ---- | ----- | ---------- |
+| user_id | snowflake | filter the log for a user id |
+| action_type | int  | the type of [audit log event](#DOCS_AUDIT_LOG/audit-log-entry-object-audit-log-events) |
+| before | snowflake | filter the log before a certain entry id |
```
this reminds me. i should really merge that before, after & around PR to add support for those values.

jhgg commented on docs/resources/Guild.md:
```diff
@@ -36,11 +39,46 @@ Guilds in Discord represent a collection of users and channels into an isolated
+| Level | Integer |
+| ----- | ------- |
+| NONE | 0 |
+| ENABLED | 1 |
```
s/ENABLED/ELEVATED/

Elevated = requires 2fa for actions that use the following permission: 
```
    KICK_MEMBERS,
    BAN_MEMBERS,
    ADMINISTRATOR,
    MANAGE_CHANNELS,
    MANAGE_GUILD,
    MANAGE_ROLES,
    MANAGE_MESSAGES,
    MANAGE_WEBHOOKS,
```

jhgg commented on docs/resources/Guild.md:
```diff
@@ -36,11 +39,46 @@ Guilds in Discord represent a collection of users and channels into an isolated
+| LOW | 1 | must have verified email on account |
+| MEDIUM | 2 | must be registered on Discord for longer than 5 minutes |
+| HIGH | 3 |  (╯°□°）╯︵ ┻━┻ - must be a member of the server for longer than 10 minutes |
+| VERY_HIGH | 4 | ┻━┻ミヽ(ಠ益ಠ)ﾉ彡┻━┻ - must have a verified phone number
```
missin a | at the end here.

jhgg commented on docs/resources/Guild.md:
```diff
@@ -161,6 +204,23 @@ Represents an Offline Guild, or a Guild whose information has not been provided
+| ALL_MESSAGES | 0
+| ONLY_MENTIONS | 1
+| NO_MESSAGES| 2
+| NULL | 3
```
remove NULL & NO_MESSAGES. those are not valid for this field.

jhgg commented on docs/resources/Guild.md:
```diff
@@ -161,6 +204,23 @@ Represents an Offline Guild, or a Guild whose information has not been provided
 | require_colons | bool | whether this emoji must be wrapped in colons |
 | managed | bool | whether this emoji is managed |
 
+###### Default Message Notification Level
```
this is duped from line 42! de-ja vu?!

msciotti replied:
Weird—didn't delete in the other place. Fixed

jhgg commented on docs/topics/RPC.md:
```diff
@@ -413,9 +413,9 @@ Used to get a channel the client is in.
-| bitrate | Int | (voice) bitrate of voice channel |
-| user_limit | Int | (voice) user limit of voice channel (0 for none) |
-| position | Int | position of channel in channel list |
+| bitrate | integer | (voice) bitrate of voice channel |
```
why lower case i when everything else is uppercase? what's the difference between integer and Number?

msciotti replied:
I've decided to undo all the changes in RPC.md. There's a lot that needs to be worked on there, so I'll do a separate commit with the fixes for that page

msciotti committed:
Renamed Audit Log Object Structure
Fixed MFA levels
Removed invalid message notification levels
Removed duplicate tables
Fixed missing pipes

msciotti committed:
Added changelog entry for audit log documentation

msciotti committed:
Merge branch 'master' into feature/AuditLogDocumentation
